examples/cpp/grpc/pingpong/pingpong.cpp

```diff
@@ -0,0 +1,276 @@
+#include "pingpong_grpc.h"
+#include "pingpong_types.h"
+
+// countdown_event.h and event.h needed for test purposes
+#include <bond/ext/detail/countdown_event.h>
+#include <bond/ext/detail/event.h>
+
+#include <bond/ext/grpc/io_manager.h>
+#include <bond/ext/grpc/server.h>
+#include <bond/ext/grpc/server_builder.h>
+#include <bond/ext/grpc/unary_call.h>
+#include <bond/ext/grpc/thread_pool.h>
+
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::ServerBuilder;
+using grpc::Status;
+using grpc::StatusCode;
+
+using bond::ext::detail::countdown_event;
+using bond::ext::detail::event;
+using bond::ext::gRPC::io_manager;
+
+using namespace pingpong;
+
+event pingNoResponse_event;
+
+class DoublePingServiceImpl final : public DoublePing::Service
+{
+    void Ping(
+        bond::ext::gRPC::unary_call<
+        bond::bonded<PingRequest>,
+        bond::bonded<PingReply>> call) override
+    {
+        PingRequest request = call.request().Deserialize();
+
+        PingReply reply;
+        reply.message = "ping " + request.name;
+
+        call.Finish(bond::bonded<PingReply>{reply}, Status::OK);
+    }
+
+    void PingNoPayload(
+        bond::ext::gRPC::unary_call<
+        bond::bonded<bond::Void>,
+        bond::bonded<PingReply>> call) override
+    {
+        PingReply reply;
+        reply.message = "ping pong";
+
+        call.Finish(bond::bonded<PingReply>{reply}, Status::OK);
+    }
+
+    void PingNoResponse(
+        bond::ext::gRPC::unary_call<
+        bond::bonded<PingRequest>,
+        bond::bonded<bond::Void>> call) override
+    {
+        PingRequest request = call.request().Deserialize();
+
+        call.Finish(bond::bonded<bond::Void>{bond::Void()}, Status::OK);
```

Comment:
We should add a comment like this here:
```
// TODO: the current implementation requires that we respond with dummy data. This will be fixed in a later release.
```